sympy/core/numbers.py

```diff
@@ -211,6 +211,7 @@ def igcd(*args):
 except ImportError:
     def igcd2(a, b):
         """Compute gcd of two Python integers a and b."""
+        a, b = as_int(a), as_int(b)
```

Comment:
Do you recall the nature of the failing test? As you say, perhaps the recasting should happen in the function that was calling `igcd2`.